pkg/loader/plugin_loader.c

```diff
@@ -60,152 +61,166 @@ static inline size_t plugin_loader_strlcpy(char *dst, const char *src, size_t si
+		prefix_len += strlen(sep);
+	}
+	plugin_loader_strlcpy(&tmp[prefix_len], s, PLUGIN_MAX_ERRLEN - prefix_len);
+	plugin_loader_strlcpy(s, tmp, PLUGIN_MAX_ERRLEN);
 }
 
-static inline void err_append(char* s, const char* suffix, const char* sep)
-{
-    if (*s != '\0')
-    {
-        strncat(s, sep, PLUGIN_MAX_ERRLEN - strlen(sep));
-    }
-    strncat(s, suffix, PLUGIN_MAX_ERRLEN - strlen(suffix));
+static inline void err_append(char* s, const char* suffix, const char* sep) {
+	if(*s != '\0') {
+		strlcat(s, sep, PLUGIN_MAX_ERRLEN);
+	}
+	strlcat(s, suffix, PLUGIN_MAX_ERRLEN);
 }
 
-static void* getsym(library_handle_t handle, const char* name)
-{
+static void* getsym(library_handle_t handle, const char* name) {
 #ifdef _WIN32
-	return (void*) GetProcAddress(handle, name);
+	return (void*)GetProcAddress(handle, name);
 #else
-	return (void*) dlsym(handle, name);
+	return (void*)dlsym(handle, name);
 #endif
 }
 
 // little hack for simplifying the plugin_load function
-#define SYM_RESOLVE(h, s) \
-    *(void **)(&(h->api.s)) = getsym(h->handle, "plugin_"#s)
-
-plugin_handle_t* plugin_load(const char* path, char* err)
-{
-    // alloc and init memory
-    err[0] = '\0';
-    plugin_handle_t* ret = (plugin_handle_t*) calloc (1, sizeof(plugin_handle_t));
-    if (!ret)
-    {
-        plugin_loader_strlcpy(err, "error allocating plugin handle", PLUGIN_MAX_ERRLEN);
-        return NULL;
-    }
-
-    // open dynamic library
+#define SYM_RESOLVE(h, s) *(void**)(&(h->api.s)) = getsym(h->handle, "plugin_" #s)
+
+plugin_handle_t* plugin_load(const char* path, char* err) {
+	// alloc and init memory
+	err[0] = '\0';
+	plugin_handle_t* ret = (plugin_handle_t*)calloc(1, sizeof(plugin_handle_t));
+	if(!ret) {
+		plugin_loader_strlcpy(err, "error allocating plugin handle", PLUGIN_MAX_ERRLEN);
+		return NULL;
+	}
+
+	// open dynamic library
 #ifdef _WIN32
-    ret->handle = LoadLibrary(path);
-    if(ret->handle == NULL)
-    {
-        DWORD flg = FORMAT_MESSAGE_ALLOCATE_BUFFER
-            | FORMAT_MESSAGE_FROM_SYSTEM
-            | FORMAT_MESSAGE_IGNORE_INSERTS;
-        LPTSTR msg_buf = 0;
-        if (FormatMessageA(flg, 0, GetLastError(), 0, (LPTSTR) &msg_buf, 0, NULL) && msg_buf)
-        {
-            plugin_loader_strlcpy(err, msg_buf, PLUGIN_MAX_ERRLEN);
-            LocalFree(msg_buf);
-        }
-    }
+	ret->handle = LoadLibrary(path);
+	if(ret->handle == NULL) {
+		DWORD flg = FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM |
+		            FORMAT_MESSAGE_IGNORE_INSERTS;
+		LPTSTR msg_buf = 0;
+		if(FormatMessageA(flg, 0, GetLastError(), 0, (LPTSTR)&msg_buf, 0, NULL) && msg_buf) {
+			plugin_loader_strlcpy(err, msg_buf, PLUGIN_MAX_ERRLEN);
+			LocalFree(msg_buf);
+		}
+	}
 #else
-    ret->handle = dlopen(path, RTLD_LAZY|RTLD_DEEPBIND);
-    if (ret->handle == NULL)
-    {
-        plugin_loader_strlcpy(err, (const char*) dlerror(), PLUGIN_MAX_ERRLEN);
-    }
+	ret->handle = dlopen(path, RTLD_LAZY);
```

Comment:
@geraldcombs FYI We have just discovered this line change involuntarily removed the fix https://github.com/falcosecurity/plugin-sdk-go/pull/104

cc @ekoops @jasondellaluce